.travis.yml

```diff
@@ -25,41 +43,74 @@ install:
   - conda config --append channels conda-forge
   # Useful for debugging any issues with conda
   - conda info -a
-  # Test conda build
-  - conda build -q conda-recipe --python $TRAVIS_PYTHON_VERSION --output-folder bld-dir
-  - conda config --add channels "file://`pwd`/bld-dir"
-  #Grab all dependencies
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pcdsdevices --file dev-requirements.txt
-  #Launch Conda environment
-  - source activate test-environment
+  # Test conda build and create test environment
+  - |
+    if [[ $UNIT_TEST || $BUILD_DOCS ]]; then
+      echo "Building full environment"
+      conda build -q conda-recipe --python $TRAVIS_PYTHON_VERSION --output-folder bld-dir
+      conda config --add channels "file://`pwd`/bld-dir"
+      conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pcdsdevices --file dev-requirements.txt
+    elif [[ $STYLE ]]; then
+      echo "Building minimal flake8 environment"
+      conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION flake8
+    fi
+  # Launch Conda environment
+  - conda activate test-environment
+  # Add the docs packages
+  - |
+    if [[ $BUILD_DOCS ]]; then
+      echo "Adding docs requirements"
+      conda install --file docs-requirements.txt
+    fi
 
 script:
-  - coverage run run_tests.py
-  - coverage report -m
-  - flake8 pcdsdevices/*.py
-  - flake8 tests/*.py
-  - set -e
+  # Run tests
+  - |
+    if [[ $UNIT_TEST ]]; then
+      echo "Running tests"
+      coverage run run_tests.py
+      coverage report -m
+    fi
+  # Check style
+  - |
+    if [[ $STYLE ]]; then
```

Comment:
I like the `$STYLE` variable for its simplicity but think it may fall short of what we'd want when breaking it out into our pcds-ci-helpers. That's why I went with the following pattern: https://github.com/pcdshub/pcds-ci-helpers/blob/master/travis/python_linter.sh

`init.sh` calls that script, so a copy/pasted `git clone`/`source init.sh` + a build matrix entry is sufficient to take care of a few scenarios.

Reply:
Yeah, this seems correct too. So for this repo `$LINT_PYTHON` would be set to `pcdsdevices/*.py tests/*.py`

Reply:
Or just set it to `pcdsdevices tests` and set the `flake8` ignores config properly